app/models/concerns/audit_extensions.rb

```diff
@@ -98,7 +100,19 @@ def set_taxonomies
         self.organization_ids = auditable.organization_ids
       elsif auditable.is_a? Organization
         self.organization_ids = [auditable_id]
+      elsif associated
+        set_taxonomies_using_associated('organization')
       end
     end
   end
+
+  def set_taxonomies_using_associated(key_name)
+    ids_arr = []
+    if associated.respond_to?(:"#{key_name}_id")
+      ids_arr = [associated.send("#{key_name}_id")].compact.uniq
+    elsif associated.respond_to?(:"#{key_name}_ids")
+      ids_arr = associated.send("#{key_name}_ids")
+    end
+    self.send("#{key_name}_ids=", ids_arr)
```

Comment:
Is there a way that the `#{key_name}_id`, `"#{key_name}_ids"` and `#{key_name}_ids=` does not exists at all?

Reply:
In `set_taxonomies` method, we already checked for `SETTINGS[:organizations_enabled] &  SETTINGS[:locations_enabled]`. 
`#{key_name}_ids=` method called on `audit` object so this method will exist for audit record.

For associated object it is good to check for methods i.e ` #{key_name}_id, "#{key_name}_ids"`.

@tbrisker, please correct me if I am missing anything.

Reply:
What @kgaikwad said, we don't know if the associated object has 0,1, or many taxonomies which is why the `respond_to?` check is needed, but we know that audit does so `#{key_name}_ids=` is defined.

Reply:
@tbrisker  the thing that captured my eyes here, is that if both `if` validation fails, it still send the `self.send`, so I was, but if `respond_to` failed, then the writer should also not exists

Reply:
note that `respond_to?` is checked on `associated`, while `self` is the audit object, not the associated one.

Reply:
right, sorry